src/client/src/commons/auth/CognitoLogoutHandler.js

```diff
@@ -0,0 +1,30 @@
+import { useEffect } from "react";
+
+// AWS Cognito currently does not support Client initiated logout according to OIDC standard:
+// https://openid.net/specs/openid-connect-rpinitiated-1_0.html
+// This component replaces the logout logic of userManager to implement call cognito logout endpoint non-standard
+// client_id and logout_uri parameters. See: https://docs.aws.amazon.com/cognito/latest/developerguide/logout-endpoint.html
+export const CognitoLogoutHandler = ({ userManager }) => {
+  useEffect(() => {
+    userManager.metadataService.getMetadata().then(metadata => {
+      if (metadata.end_session_endpoint.includes("amazoncognito.com")) {
```

Comment:
Q: Stand die `end_session_endpoint` URL immer schon in der OpenID Konfiguration? Haben wir die übersehen? Oder konntest du das reinkonfigurieren? Jetzt macht es IMHO noch weniger Sinn, dass Cognito die URL in die OpenID Konfig schreibt, dann aber nicht OIDC konform Logout macht 🤔

Reply:
Seit Dienstag morgen 27.02.24 steht der `end_session_endpoint` im OpenID Discovery-Dokument vom Cognito. Leider ist der Endpoint weiterhin nicht nach reinem OIDC Konform, bzw. muss konfiguriert werden. 
Bei oidc-client-ts gibts dazu auch ein [Issue #1385](https://github.com/authts/oidc-client-ts/issues/1385). Kann man sich wohl leisten, wenn man ein grosser Anbieter ist. 🤷